installer/bash-installer

```diff
@@ -0,0 +1,212 @@
+
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+set -euo pipefail
+
+CASTOR_EXECUTABLE="castor"
+CASTOR_TMP_NAME="$CASTOR_EXECUTABLE-"$(date +"%s")
+CASTOR_NAME="🦫  Castor"
+CASTOR_DOWNLOAD_URL_PATTERN="https://github.com/jolicode/castor/releases/latest/download/castor.~platform~.phar"
+CASTOR_TMPDIR="${TMPDIR:-/tmp}"
+
+function output {
+    style_start=""
+    style_end=""
+    if [ "${2:-}" != "" ]; then
+    case $2 in
+        "success")
+            style_start="\033[0;32m"
+            style_end="\033[0m"
+            ;;
+        "error")
+            style_start="\033[31;31m"
+            style_end="\033[0m"
+            ;;
+        "info"|"warning")
+            style_start="\033[33m"
+            style_end="\033[39m"
+            ;;
+        "heading")
+            style_start="\033[1;33m"
+            style_end="\033[22;39m"
+            ;;
+    esac
+    fi
+
+    builtin echo -e "${style_start}${1}${style_end}"
+}
+
+output "${CASTOR_NAME} installer" "heading"
+
+binary_dest="${HOME}/.local/bin"
+custom_dir="false"
+
+# Getops does not support long option names
+while [[ $# -gt 0 ]]; do
+case $1 in
+    --install-dir=*)
+        binary_dest="${1#*=}"
+        custom_dir="true"
+        shift # past argument=value
+        ;;
+    --install-dir)
+        binary_dest="${2:-}"
+        custom_dir="true"
+        shift # past argument
+        shift # past value
+        ;;
+    *)
+        output "Unknown option $1" "error"
+        output "Usage: ${0} [--install-dir=dir]"
+        exit 1
+        ;;
+esac
+done
+
+# Run environment checks.
+output "\nEnvironment check" "heading"
+
+# Check that cURL or wget is installed.
+downloader=""
+if command -v curl >/dev/null 2>&1; then
+    downloader="curl"
+    output "  [*] cURL is installed" "success"
+elif command -v wget >/dev/null 2>&1; then
+    downloader="wget"
+    output "  [*] wget is installed" "success"
+else
+    output "  [ ] ERROR: cURL or wget is required for installation" "error"
+    exit 1
+fi
+
+# Check that PHP is installed.
+if command -v php >/dev/null 2>&1; then
+    output "  [*] PHP is installed" "success"
+else
+    output "  [ ] ERROR: PHP is required for installation" "error"
+    exit 1
+fi
+
+if [[ "1" == "$(php -r 'echo version_compare(PHP_VERSION, "8.1", ">=");')" ]]; then
+    output "  [*] PHP version >= 8.1" "success"
+else
+    output "  [ ] ERROR: PHP version is not 8.1 or greater" "error"
```

Comment:
Should we display the current detected PHP version (in both success and error cases)? Fe in case the user has several versions installed, it could help to understand the issue. WDYT ?

Reply:
I agree 👍 
done